### ISSUE TITLE
unneeded :or bindings

### ISSUE THREAD

**Issue by johanatan** [borkdude/clj-kondo#245]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
:or bindings (coupled with :keys) could also be analyzed for usage. In general, I can see no reason for a binding to exist in an :or vector if that binding doesn't need to be in its associated :keys vector.

**Comment by borkdude** [comment]
@johanatan Can you provide an example to clarify this feature to make sure I understand correctly? This feature already exists in clj-kondo:

<img width="496" alt="Screenshot 2019-06-10 08 35 35" src="https://user-images.githubusercontent.com/284934/59176945-d4803400-8b5a-11e9-8ec0-fb975cb8443a.png">

**Comment by johanatan** [comment]
Suppose :bar were unused and marked as such by clj-kondo. If :bar also appeared in the :or map, would there be a warning for it? Pretty sure I didn’t see one earlier.

**Comment by borkdude** [comment]
@johanatan Underscoring a binding like:

```
(let [{:keys [:_foo :bar :baz]
       :or {_foo 1}}])
```
is telling the linter: don't report about this binding, ignore. That's not the same as telling: this binding is unused. Because you can still use it in Clojure.

**Comment by borkdude** [comment]
@johanatan I see.

**Comment by borkdude** [comment]
This will work as follows: the users sees that bar is unused as reported, then removes it, and then clj-kondo will report about or being a default defined for an unbounded named. The reason why the name is not reported as an unused binding is that or doesn't introduce new bindings.

**Comment by johanatan** [comment]
I didn’t ask for it to be reported as an unused binding. I was suggesting for it to be reported as unnecessary *when* the same key is also simultaneously being reported as unused in the context.

**Comment by johanatan** [comment]
Why does it matter if :or “creates a new binding” or not? It’s clearly unused/unneeded/unnecessary in the situation I describe and thus deserves to be reported. IMHO.

Why make users iterate more than they have to?

**Comment by borkdude** [comment]
@johanatan I see. The previous comment was written from the POV of editor usage where new warnings usually appear in milliseconds, but from the POV of CI usage it makes sense. I'll have a look.

**Comment by johanatan** [comment]
Ah, gotcha. Yea I’m currently using as CLI for the time being.

**Comment by borkdude** [comment]
@johanatan Thanks to @OscarMarshall who fixed this:

<img width="437" alt="Screenshot 2020-04-08 11 15 22" src="https://user-images.githubusercontent.com/284934/78766976-4a774680-798a-11ea-8fe2-e69c1ea7dbc4.png">
